packages/belos/src/BelosGmresPolyOp.hpp

```diff
@@ -861,18 +858,18 @@ namespace Belos {
       //For each value, compute (a log of) a product of differences:
       for(int i = 0; i < dimN; i++) 
       {
-        prod(i,0) = MCT::one();
+        prod(i) = MCT::one();
         for(int k = 0; k < j; k++)
         {
           a = thetaN(i,0) - sorted(k,0);
           b = thetaN(i,1) - sorted(k,1);
-          prod(i,0) = prod(i,0) + log10(sqrt(a*a + b*b));
+          prod(i) = prod(i) + log10(sqrt(a*a + b*b));
         }
       }
       
       //Value with largest product goes in the next slot:
       MagnitudeType * maxPointer = std::max_element(prod.values(), (prod.values()+dimN));
-      int maxIndex = (maxPointer- prod.values());
+      int maxIndex = int (maxPointer- prod.values());
```

Comment:
This could be a problem if none of them are largest.  You mentioned that the sequence is nonempty by this point, so the only way that could happen is if all entries are NaN.  You'll have other problems if that's the case, though ;-) .